controllers/snapshot/snapshot_adapter.go

```diff
@@ -75,6 +76,91 @@ func scenariosNamesToList(integrationTestScenarions *[]v1beta1.IntegrationTestSc
 	return &result
 }
 
+// EnsureRerunPipelineRunsExist is responsible for recreating integration test pipelines triggered by users
+func (a *Adapter) EnsureRerunPipelineRunsExist() (controller.OperationResult, error) {
+
+	scenarioName, ok := gitops.GetIntegrationTestRunLabelValue(*a.snapshot)
+	if !ok {
+		// no test rerun trigerred
+		return controller.ContinueProcessing()
+	}
+	integrationTestScenario, err := a.loader.GetScenario(a.client, a.context, scenarioName, a.application.Namespace)
+
+	if err != nil {
+		if clienterrors.IsNotFound(err) {
+			a.logger.Error(err, "scenario for integration test re-run not found", "scenario", scenarioName)
+			// scenario doesn't exist just remove label and continue
+			if err = gitops.RemoveIntegrationTestRerunLabel(a.client, a.context, a.snapshot); err != nil {
+				return controller.RequeueWithError(err)
+			}
+			return controller.ContinueProcessing()
+		}
+		return controller.RequeueWithError(fmt.Errorf("failed to fetch requested scenario %s: %w", scenarioName, err))
+	}
+
+	a.logger.Info("Re-running integration test for scenario", "scenario", scenarioName)
+
+	testStatuses, err := gitops.NewSnapshotIntegrationTestStatusesFromSnapshot(a.snapshot)
+	if err != nil {
+		return controller.RequeueWithError(err)
+	}
+
+	integrationTestScenarioStatus, ok := testStatuses.GetScenarioStatus(integrationTestScenario.Name)
+	if ok && integrationTestScenarioStatus.Status == intgteststat.IntegrationTestStatusInProgress {
```

Comment:
Thinking about this, maybe we also want to include the pending state here? 
If our re-reconciliation upon failing to remove the label starts before the PLR starts, we might find the status still in pending?

Reply:
To clarify, I'm thinking about the ephemeral environment scenario here, the static example will be in the `InProgress` state.

Reply:
That makes sense, added